lib/src/model.dart

```diff
@@ -2163,21 +2135,32 @@ abstract class ModelElement extends Nameable
         newModelElement = new Typedef(e, library);
       }
       if (e is FieldElement) {
+        assert(getter != null || setter != null);
         if (enclosingClass == null) {
-          if (index != null) {
-            assert(getter != null);
-            newModelElement =
-                new EnumField.forConstant(index, e, library, getter);
-          } else {
-            if (e.enclosingElement.isEnum) {
-              newModelElement = new EnumField(e, library, getter, setter);
+          if (e.enclosingElement.isEnum) {
+            int index =
+                e.computeConstantValue()?.getField('index')?.toIntValue();
```

Comment:
This definitely works but it took a bit of thinking before I understood why.  Essentially what's happening here is that the fields in an enum class `E` are all static fields having type `E`, except for the two "special" fields `index` (a non-static field having type `int`) and `values` (a static field having type  `List<E>`).  You only want your new behavior for the non-special fields, and it works because only the type `E` has a non-static member `index` whose value is an `int`; the types `int` and `List<E>` don't.

If I'm understanding the code correctly there's a getter `FieldElement.isEnumConstant` that does what you want.  So I think you could replace lines 2140-2161 with something like:

```dart
if (e.isEnumConstant) {
  int index = e.computeConstantValue().getField('index').toIntValue();
  newModelElement = new EnumField.forConstant(index, e, library, getter);
} else if (e.enclosingElement.isEnum) {
  newModelElement = new EnumField(e, library, getter, setter);
} else {
  newModelElement = new Field(e, library, getter, setter);
}
```